### ISSUE TITLE
Add option to show deprecated apis and fields

### ISSUE THREAD

**Issue by marsom** [instrumenta/kubeval#189]
A nice feature whould be an option to show deprecated api and fields, i.e. --show-deprecations.

Use case: I want to know if I use any deprecated stuff in my resources.

My simple and not so correct way to identify if I use any deprecated API is: I have a clone of the schema repo, and I delete all schemas which contain the string "deprecated" in the first to line. kubescore cannot find any schema for these apis and complains.

**Comment by swade1987** [comment]
This would be super useful as when you add `-v 1.16.0` you should expect failures aligning with https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/

**Comment by balteravishay** [comment]
Agree on expecting failure on deprecated apis when using -v 1.16.0. is this in the roadmap?

**Comment by marsom** [comment]
https://github.com/swade1987/deprek8ion

**Comment by nicorikken** [comment]
I came here looking for an explanation why I couldn't get errors on deprecated API's. My use-case for kubeval is to detect deprecation issues before doing a cluster-upgrade. It seems kubeval is currently not suitable for this job, right?

**Comment by marsom** [comment]
it's not suitable, that's correct. 

however you could use https://github.com/swade1987/deprek8ion or wait for kubernetes v1.19.0 which will show any deprecated api usages, see https://github.com/kubernetes/kubernetes/pull/92842 for details.

**Comment by nicorikken** [comment]
@marsom I couldn't get deprek8ion to error on deprecations. Perhaps you can show me an example?

I've done some thinking and searching:

It seems part of the issue is in the used source for API specs. For example the `extensions/v1beta1/policy` API has been deprecated for version `1.16`: https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/ but is still included in the [spec source, even for version 1.17.4](https://github.com/instrumenta/kubernetes-json-schema/blob/master/v1.17.4/podsecuritypolicy-extensions-v1beta1.json). The term 'deprecated' is used in the description though.

Some close to working solutions I can come up with:

**1. Expand schemas**
Work with [instrumenta/kubernetes-json-schema](https://github.com/instrumenta/kubernetes-json-schema) to include a parameter to flag if an API is considered deprecated. This makes is more explicit. I'm not sure what this implies in terms of schema compatibility.

_I think this would be the best solution, making it very explicit._

**2. Check on deprecated in the description**
When parsing the schema, check the description for the word deprecated to filter out deprecated. And have a flag to toggle this behavior.

_I consider this a second best, as it can be executed with the current schemas, but relies on text-parsing._

**3. Mirror the API schemas**
Download the API-schemas, grep on 'Deprecated' and remove those schemas from that location. The point to that directory using the `--schema-location` parameter. In case of an error, that would throw:
```
ERR  - stdin: Failed initializing schema https://mymirror/v1.17.1-standalone/podsecuritypolicy-extensions-v1beta1.json: Could not read schema from HTTP, response status is 404 Not Found
``` 

_This is something that one could could quite easily do themselves to get validation here and now._

**4. Allow API's in the skip-kinds parameter**
Currently one can use the `--skip-kinds` flag to exclude certain kinds. For API-deprecation specifically this is no good, as the kind remains the same, it is the apiVersion that changes.
So instead of:
```
cat test-psp.yaml | kubeval -v 1.17.1 --reject-kinds PodSecurityPolicy
```
One could do:
```
cat test-psp.yaml | kubeval -v 1.17.1 --reject-kinds 'extensions/v1beta1/podSecurityPolicy'
ERR  - Prohibited resourse kind 'extensions/v1beta1/PodSecurityPolicy' in stdin
```

_I think this is a good improvement anyway, but requires a user to concatenate all deprecations to get a more strict validation._

**Comment by marsom** [comment]
create a file which looks like this:

```
pluto:~ marsom$ cat test.yaml 
---
apiVersion: extensions/v1beta1
kind: PodSecurityPolicy
metadata:
  name: restrictive-psp
spec:
  privileged: false
  # Required to prevent escalations to root.
  allowPrivilegeEscalation: false
```
then test it

```
pluto:~ marsom$ cat ./test.yaml | docker run --rm -i eu.gcr.io/swade1987/deprek8ion:1.1.17 -
FAIL - PodSecurityPolicy/restrictive-psp: API extensions/v1beta1 for PodSecurityPolicy has been deprecated, use policy/v1beta1 instead.

16 tests, 15 passed, 0 warnings, 1 failure
```
